fix(checkbox): fix checkbox when modelvalue length less than min limit

cokemine opened the pull request:
fix #1742

isLimitExceeded shouldn't be able to limit by min.
This may cause bug when model length < min - 1.

Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer to relative issues for your PR.

cokemine committed:
fix(checkbox): fix checkbox when modelvalue length less than min limit

JeremyWuuuuu commented on packages/checkbox/src/useCheckbox.ts:
```diff
@@ -50,9 +50,6 @@ const useModel = (props: ICheckboxProps) => {
       if (isGroup.value && Array.isArray(val)) {
         isLimitExceeded.value = false
 
-        if (checkboxGroup.min !== undefined && val.length < checkboxGroup.min.value) {
-          isLimitExceeded.value = true
-        }
```
Changes here makes sense, but we need to find a way to utilize `min` property which should report error to the upper layer like `el-form` or somewhere else, we need to let the users know that the limit has not been reached.

JeremyWuuuuu replied:
I think the value here `isLimitExceeded` was wrong in the first place, because setting the `min` property will never get the limitation exceeded, it should be something like `hasNotReached` flag, anyways, we need to find a way to report the error. 
I need to to work on that.

CokeMine replied:
Yes. min should not get limitation exceeded.  El-form has it's own validation. I think it can be validated by the validater from elform

JeremyWuuuuu replied:
Yeah, the `min` validation should be checked when its wrapped by `el-form`.

CokeMine commented:
@JeremyWuuuuu Hi, Will this PR be merged? This PR only solved one problem that `isLimitExceeded` was limited by `min`.If so, I will add some tests and mark it as `Ready for review`.
I think it will get a big change if validate `min` automatically when wrapped by `el-form` and wait for your work.

Now it can be validated manually such as,
```
data() {
    return {
      Form: {
        cities: [],
      },
      cityOptions,
      rules: {
        city: {
          message: '不合法',
          validator: () => this.Form.cities.length > 2 && this.Form.cities.length < 5,
          trigger: 'change',
        },
      },
    }
  }

sxzz commented:
Hi, sorry for delay so long.
Could you please resolve conflicts? We changed file structure a lot recently.

cokemine commented:
> Hi, sorry for delay so long.
> Could you please resolve conflicts? We changed file structure a lot recently.

Sorry I almost forgot about it. This PR does not completely fix the issue, if  need to fix it, may need to use `provide` and `inject`.
However I still think the code I deleted should be removed.

cokemine committed:
fix

cokemine committed:
fix something wrong while handling conflict
